packages/gafl-webapp-service/src/pages/multibuy/remove-licence/__tests__/update-transaction.spec.js

```diff
@@ -85,13 +96,22 @@ describe('remove-licence > update transaction', () => {
     expect(mockRequest.cache().helpers.status.setCurrentPermission()).toEqual(expect.objectContaining(setPermission))
   })
 
-  it('transaction.set is being called with a permission removed', async () => {
+  it('transaction.set is being called with a permission removed with no duplciates', async () => {
+    hasDuplicates.mockReturnValueOnce(false)
     const transaction = { permissions: [getTransactionPermissionOne(), getTransactionPermissionTwo(), getTransactionPermissionThree()] }
     const mockRequest = createMockRequest({ cache: { transaction: transaction } })
     await updateTransaction(mockRequest)
     expect(mockRequest.cache().helpers.transaction.set()).toEqual(expect.objectContaining(transaction))
   })
 
+  it('transaction.set is being called with a permission removed with duplciates', async () => {
```

Comment:
I think you need to improve this test description. It should read 'when removing one of several identical permissions, only one of them should be removed and the others should remain'.

I think you also need to beef this test up into an it.each, where you try different cases:
1. The case you're testing at the moment with _just_ two identical permissions
2. Several permissions, all different apart from two identical permissions that aren't consecutively placed
3. Several permissions, all different apart from three identical permissions that aren't consecutively placed